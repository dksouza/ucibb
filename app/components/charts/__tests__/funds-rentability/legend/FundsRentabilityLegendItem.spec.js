import React from 'react';
import { shallow } from 'enzyme';
import FundsRentabilityLegendItem from '../../../funds-rentability/legend/legendItem';

let wrapper = {};
let props = {};

describe('FundsRentabilityLegendItem', () => {
  beforeAll(() => {
    props = {
      fundName: 'MULTI DINAMICO V',
      percent: 9.09,
      yieldValue: 0,
      value: 277.3,
      index: 1,
    };

    return props;
  });

  beforeEach(() => {
    wrapper = shallow(<FundsRentabilityLegendItem {...props} />);
    return wrapper;
  });

  it('Shallow View', () => {
    const newProps = { ...props, index: 0 };
    const newWrapper = shallow(<FundsRentabilityLegendItem {...newProps} />);
    expect(newWrapper.find('View').exists()).toBe(true);
  });

  it('Shallow FundPercentage', () => {
    expect(wrapper.find('FundPercentage').exists()).toBe(true);
  });

  it('Shallow BBCurrencyValue', () => {
    expect(wrapper.find('BBCurrencyValue').exists()).toBe(true);
  });

  it('Shallow BBText', () => {
    expect(wrapper.find('BBText').exists()).toBe(true);
  });

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
