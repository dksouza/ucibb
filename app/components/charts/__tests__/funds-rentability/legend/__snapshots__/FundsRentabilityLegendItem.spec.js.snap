// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundsRentabilityLegendItem Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginHorizontal": 32,
      },
      Object {
        "marginTop": 15,
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginTop": 3,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 50,
            "height": 16,
            "width": 16,
          },
          Object {
            "backgroundColor": "#00ACC1",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "marginLeft": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <FundPercentage
        name="MULTI DINAMICO V"
        percent={9.09}
      />
      <BBCurrencyValue
        big={false}
        extraSmall={false}
        light={false}
        normal={true}
        small={false}
        style={
          Object {
            "alignSelf": "flex-end",
          }
        }
        textAfterValue=""
        theme="dark"
        value={277.3}
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <BBText
        accessibilityLabel="Quanto já rendeu? 0"
        regular={true}
        style={
          Object {
            "color": "rgba(0,0,0,0.54)",
            "paddingTop": 0,
          }
        }
        tinyText={true}
      >
        Quanto já rendeu?
      </BBText>
      <BBText
        accessibilityLabel="Valor: 0"
        regular={true}
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "rgba(0,0,0,0.54)",
            "paddingTop": 0,
          }
        }
        tinyText={true}
      >
        R$ 0,00
      </BBText>
    </View>
  </View>
</View>
`;
